onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2674,6 +2672,282 @@ Status LRNOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
 
 #pragma endregion
 
+#pragma region op_clip
+
+class ClipOpBuilder : public BaseOpBuilder {
+ public:
+  void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) override;
+
+ private:
+  bool IsOpSupportedImpl(ModelBuilder& model_builder, const Node& node) override;
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node) override ORT_MUST_USE_RESULT;
+  static bool GetMinMax(ModelBuilder& model_builder, const Node& node, float& min, float& max);
+};
+
+void ClipOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) {
+  if (node.InputDefs().size() > 1)
+    model_builder.AddInitializerToSkip(node.InputDefs()[1]->Name());  // min
+
+  if (node.InputDefs().size() > 2)
+    model_builder.AddInitializerToSkip(node.InputDefs()[2]->Name());  // max
+}
+
+/* static */ bool ClipOpBuilder::GetMinMax(ModelBuilder& model_builder, const Node& node, float& min, float& max) {
+  const auto& initializers(model_builder.GetInitializerTensors());
+
+  if (node.InputDefs().size() > 1) {  // we have input min
+    const auto& min_name = node.InputDefs()[1]->Name();
+    if (!Contains(initializers, min_name)) {
+      LOGS_DEFAULT(VERBOSE) << "Input min of Clip must be known";
+      return false;
+    }
+    min = GetTensorFloatData(initializers.at(min_name))[0];
+  }
+
+  if (node.InputDefs().size() > 2) {  // we have input max
+    const auto& max_name = node.InputDefs()[2]->Name();
+    if (!Contains(initializers, max_name)) {
+      LOGS_DEFAULT(VERBOSE) << "Input max of Clip must be known";
+      return false;
+    }
+    max = GetTensorFloatData(initializers.at(max_name))[0];
+  }
+
+  return true;
+}
+
+bool ClipOpBuilder::IsOpSupportedImpl(ModelBuilder& model_builder, const Node& node) {
+  if (node.SinceVersion() < 11) {
+    LOGS_DEFAULT(VERBOSE) << "Clip only supports opset 11+";
+    return false;
+  }
+
```

Comment:
Isn't it easy to support the earlier versions as min and max were attributes?

Reply:
Added the support for Clip-6

---
In reply to: [502970742](https://github.com/microsoft/onnxruntime/pull/5427#discussion_r502970742) [](ancestors = 502970742)